### DISCUSSION TITLE
"Sort by" feature

### DISCUSSION THREAD

**Discussion by MangoIV** [flora-pm/flora-server#423] [Feature Requests]
It would be nice to have a "sort by" feature, i.e. something like 
- best match (the standard) 
- downloads
- outdatedness

and a modifier for ascending/ descending

**Comment by jappeace** [comment]
specifcally for anwsering the question of "is this package being maintained", it'd be really nice  if you know that before having to go trough compiling it and finding issues like bounds.

**Reply by TixieSalander** [reply]
For now release dates are display in the versions list in the package "single page" but indeed having the last version release date present directly in packages lists and being able to sort lists on that criteria is definitely useful and will be probably be added.

**Comment by TixieSalander** [comment]
@MangoIV I agree! @tchoutri recently mentioned this feature to me as something planned in the future so it very should be added.

At the exception of downloads since we currently can't get accurate data on that:
1. Hackage have indeed a download count but the number isn't accurate at all. It doesn't include CDN who put package in cache (and other ways of getting a package, If I remember correctly what @tchoutri told me)
2. [Other repositories](https://flora.pm/documentation/namespaces) aggregated by Flora doesn't necessary provide download data

But date related sorting on packages lists is a good thing indeed! 👍

**Reply by MangoIV** [reply]
Since flora is indexing all packages anyway, one could make download count a proportion instead of a number, so basically you would split the packages blocks of most downloaded packages and maybe give them a heat depending on in which block / proportion they are. You would have to be a bit clever because I don’t expect the distribution to be equal, I.e. a small amount of total packages will have a big proportion of total downloads and a high percentage of total packages will only have a different amount of downloads because they have been on hackage for longer (in contrast to having been downloaded by real users more often). 

But would be cool to give a ballpark, maybe with some kind of rough “popularity gauge”

**Reply by tchoutri** [reply]
@MangoIV Still that wouldn't work because I imagine that highly-requested packages stay "hot" in the CDN, thus getting few hits on Hackage. But then how do you distinguish packages that are scarcely requested and those so requested that the CDN keeps them all the time? ;-) 

Unless hackage.haskell.org is patched to get fed CDN data, download numbers will be unreliable.
(moreover, we also index Cardano and Horizon, which have no data at all, since they are static sites)

**Reply by tchoutri** [reply]
That being said a stand-alone installation of Flora could certainly do this.

**Reply by MangoIV** [reply]
> Still that wouldn't work because I imagine that highly-requested packages stay "hot" in the CDN, thus getting few hits on Hackage. But then how do you distinguish packages that are scarcely requested and those so requested that the CDN keeps them all the time?

so do you think that there’s a package with some amount of real downloads and an amount of hackage downloads, such that a package with more hackage downloads has less or equal than real downloads? Because if not that wouldn’t be a problem.

**Reply by tchoutri** [reply]
I don't understand how this connects to my above statement

**Reply by MangoIV** [reply]
In the way that you would still capture the proportions correctly like proposed in the original answer.
